### ISSUE TITLE
[BUG] Elsa Nuget Package conflicts occur while migrating from Elsa 3.3.0 rc4 to Elsa 3.3.0 rc5

### ISSUE THREAD

**Issue by jayachandra21** [elsa-workflows/elsa-core#6253]
## Description
In the production environment when Elsa version is migrated from **Elsa 3.3.0 rc4** to **Elsa 3.3.0 rc5**, getting the following packages conflicts and which is breaking the existing application which is running or required to add the additional NuGet packages.

## Steps to Reproduce

1. **Detailed Steps**: 

- Migrate the Elsa Version from 3.3.0 RC4 to 3.3.0 rc5

The following Nuget versions conflicts occurs. 
          a. Microsoft.Extensions.DependencyInjection 9.0.0
          b. Microsoft.Extensions.Logging.Abstractions 9.0.0

When included this packages which is also impacts other below packages.
- Microsoft.Extensions.Hosting.Abstractions
- System.Text.Json

This could be because of removal of **`Target framework 6.0`** in the Elsa 3.3.0 rc5 version.
Please check the "Reviewebale" comments on the below PR

[pull/6233#pullrequestreview](https://github.com/elsa-workflows/elsa-core/pull/6233#pullrequestreview-252644852)

2. **Reproduction Rate**: every time

## Expected Behavior
On removal of **`Target framework 6.0`**  it should not impact the existing application which is running.

## Actual Behavior
On migration of Elsa Nuget package Versions from 3.3.0 rc4 to 3.3.0 rc5, impacting the Nuget packages conflicts.

## Environment
- **Elsa Package Version**: Elsa 3.3.0 RC5
- **Operating System**: Windows
- **Browser and Version**: Chrome

**Comment by sfmskywalker** [comment]
Should be fixed as of build [2468](https://github.com/elsa-workflows/elsa-core/actions/runs/12572802484). Let me know if otherwise. I'm probably going to do an RC6 soon before the final release, just to be sure.

**Comment by jayachandra21** [comment]
Hello @sfmskywalker, 
I have verified this bug with following versions:

Case 1:

1. Elsa Core Version :3.3.0-preview.2468 and 3.3.0-preview.2470 
2. Elsa Studio : 3.3.0-preview.698

Detected package downgrade  error occurs for the **_Elsa.Api.Client_**
![image](https://github.com/user-attachments/assets/2fc4e7ab-deb5-4871-bfe8-d6b3e159f056)

Case 2:
So I have updated the `Elsa.Api.Client` package only with a version "3.3.0-rc5". Remaining all other Elsa packages **3.3.0-preview.2470** and ElsaStudioVersion as **3.3.0-preview.698**
Getting the Microsoft.Extensions.DependencyInjection and Microsoft.Extensions.Logging.Abstractions packages error.
![image](https://github.com/user-attachments/assets/22ed67a8-2e6c-4fb7-8db5-46b9d94d0a67)
By adding this these packages other versions conflicts still occurs.
let me know if any specific nuget packages versions should be used to verify this.

**Comment by sfmskywalker** [comment]
Please make sure to use the same version of the packages. I.e., use 3.3.0-preview.2470 in your Elsa Server project, without mixing it with 3.3.0-preview.2468.

I have tried it with your sample project for troubleshooting the SQL migrations issue, and I got no issues there. Perhaps you can share a similar project with the issues that you're seeing?

**Comment by jayachandra21** [comment]
Hello @sfmskywalker I will try with the **`Elsa 3.3.0-preview.2470`**. Can we go with `**Elsa Studio 3.3.0-preview.698**`?

**Comment by sfmskywalker** [comment]
For Elsa Studio, the latest package version is `3.3.0-rc5`.
Keep in mind that `Elsa.Api.Client` is built from the elsa-core repo, which means that its latest version is `3.3.0-preview.2470`, so you probably need to reference that version.

Without seeing the full context, I am mostly guessing.

**Comment by jayachandra21** [comment]
Hello @sfmskywalker,
Here I have attached my reproduction project for your reference.

error:
![image](https://github.com/user-attachments/assets/7766bba4-c4ec-4cb4-88b9-79ba27fae39d)
[Elsa330RC5.zip](https://github.com/user-attachments/files/18289945/Elsa330RC5.zip)

**Comment by sfmskywalker** [comment]
Thanks @jayachandra21 

**ElsaServer**

To confirm, the `ElsaServer` project builds fine. No issues there.

**ElsaStudioBlazorServer**

As for `ElsaStudioBlazorServer`, its RC5 packages reference Elsa.Api.Client RC5. So for that reason, we cannot have it reference a newer package such as `3.3.0-preview.2470`.

To make the build succeed, simply reference `Elsa.Api.Client` version `3.3.0-rc5`. After that, everything builds.

<img width="781" alt="image" src="https://github.com/user-attachments/assets/fa1dce64-c5be-45b1-996e-c85912bf21d7" />

**Comment by jayachandra21** [comment]
Greetings @sfmskywalker,
Thank you very much!

As Elsa RC6 is released, I have tried with rc6 version and found below issue and remaining all above package issues are resolved.
![image](https://github.com/user-attachments/assets/17c35072-a04c-4d51-8d8e-21d2781190ff)

Is it something, we need to add the some additional package to address this issue?
I have attached sample reprod project.
[Elsa330RC6.zip](https://github.com/user-attachments/files/18291844/Elsa330RC6.zip)

**Comment by sfmskywalker** [comment]
Thanks for the repro project - I got the same error when I ran your project. The issue is that the Blazor project references RC5 for Studio and RC6 for API client. This doesn't work, because RC5 Blazor modules depend on the RC5 API client.

So to fix the issue, just upgrade everything to RC6:

<img width="940" alt="image" src="https://github.com/user-attachments/assets/936a643f-3796-48db-91ce-acb5f4568ebd" />

After that, the application starts successfully.

**Comment by jayachandra21** [comment]
> Thanks for the repro project - I got the same error when I ran your project. The issue is that the Blazor project references RC5 for Studio and RC6 for API client. This doesn't work, because RC5 Blazor modules depend on the RC5 API client.
> 
> 
> 
> So to fix the issue, just upgrade everything to RC6:
> 
> 
> 
> <img width="940" alt="image" src="https://github.com/user-attachments/assets/936a643f-3796-48db-91ce-acb5f4568ebd" />
> 
> 
> 
> After that, the application starts successfully.
> 
> 

Thank you @sfmskywalker 
Studio packages might be building while I was testing with core rc6.
At that point studio packages was rc5 in the nugget manager.

**Comment by sfmskywalker** [comment]
Makes sense, the RC6 for studio was indeed built and deployed at a much later stage.
